src/lib/ptf-api.ts

```diff
@@ -1,33 +1,129 @@
+    version: string;
+    time: number;
+    raw: any;
+}
+
+export interface GetOverviewResponse extends PricesResponse {
+    items: Array<ItemOverview>;
+}
+
+export interface ItemOverview {
+    name: string;
+    sku: string;
+}
+
+export interface GetPricelistResponse extends PricesResponse {
+    currency?: any;
+    items?: Array<Item>;
+}
+
+export interface Item {
+    sku: string;
+    name: string;
+    source: string;
+    time: number;
+    buy: KeysMetal | null;
+    sell: KeysMetal | null;
+}
+
+export interface KeysMetal {
+    keys: number;
+    metal: number;
+}
+
+export interface GetItemLinks extends PricesResponse {
+    sku: string;
+    name: string;
+    links: Links;
+}
+
+export interface Links {
+    ptf: string;
+    mptf: string;
+    scm: string;
+    bptf: string;
+}
+
+export interface GetItemPriceResponse extends PricesResponse {
+    sku?: string;
+    name?: string;
+    currency?: string;
+    source?: string;
+    time?: number;
+    buy?: KeysMetal;
+    sell?: KeysMetal;
+    message?: string;
+}
+
+export interface GetItemHistoryResponse extends PricesResponse {
+    sku: string;
+    name: string;
+    source: string;
+    currency: any;
+    history: Array<History>;
+}
+
+export interface History {
+    time: number;
+    buy: KeysMetal;
+    sell: KeysMetal;
+}
+
+export interface GetItemSalesResponse extends PricesResponse {
+    sku: string;
+    name: string;
+    sales: Array<Sale>;
+}
+
+export interface Sale {
+    id: string;
+    steamid: string;
+    automatic: boolean;
+    attributes: any;
+    intent: number;
+    currencies: KeysMetal;
+    time: number;
+}
+
+export function getSchema(): Promise<GetSchemaResponse> {
     return apiRequest('GET', '/schema', { appid: 440 });
 }
 
-export function getPricelist(source: string): Promise<UnknownDictionary<any>> {
+export function getPricelist(source: string): Promise<GetPricelistResponse> {
     return apiRequest('GET', '/items', { src: source });
 }
 
-export function getPrice(sku: string, source: string): Promise<UnknownDictionary<any>> {
+export function getPrice(sku: string, source: string): Promise<GetItemPriceResponse> {
     return apiRequest('GET', `/items/${sku}`, { src: source });
 }
 
-export function getPriceHistory(sku: string, source: string): Promise<UnknownDictionary<any>> {
+export function getPriceHistory(sku: string, source: string): Promise<GetItemHistoryResponse> {
     return apiRequest('GET', `/items/${sku}/history`, { src: source });
 }
 
-export function getSales(sku: string, source: string): Promise<UnknownDictionary<any>> {
+export function getSales(sku: string, source: string): Promise<GetItemSalesResponse> {
     return apiRequest('GET', `/items/${sku}/sales`, { src: source });
 }
 
-export function requestCheck(sku: string, source: string): Promise<UnknownDictionary<any>> {
+export function requestCheck(sku: string, source: string): Promise<PricesResponse> {
```

Comment:
This type might be wrong.

https://github.com/TF2Autobot/tf2autobot/blob/30cc4416c68c2fe435e590efaee4539b2a0ea01c/src/classes/Commands/Commands.ts#L1057

https://github.com/TF2Autobot/tf2autobot/blob/30cc4416c68c2fe435e590efaee4539b2a0ea01c/src/classes/MyHandler/offer/accepted/updateListings.ts#L40

Reply:
thanks I'll fix these

Reply:
I believe it is fixed now